src/internal/coredb/projects.go

```diff
@@ -0,0 +1,176 @@
+package coredb
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/jmoiron/sqlx"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+)
+
+// ProjectIterator batches a page of projectRow entries. Entries can be retrieved using iter.Next().
+type ProjectIterator struct {
+	lock     sync.Mutex
```

Comment:
Iterators are serial in nature.  Concurrent access would not preserve order.  It's probably safe to assume that all access will be single threaded.